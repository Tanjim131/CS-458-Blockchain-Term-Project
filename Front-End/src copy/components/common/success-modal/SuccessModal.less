.success-modal {
    top: 0;
    .modal-content-wrapper {
        text-align: center;
        .check-circle-icon {
            font-size: 32px;
            color: #52c41a;
        }
    }
}