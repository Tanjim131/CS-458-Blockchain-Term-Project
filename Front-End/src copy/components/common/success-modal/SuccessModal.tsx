import React, { FC } from "react";
import { Modal, Space } from "antd";
import { CheckCircleFilled } from "@ant-design/icons";

import "./SuccessModal.less";

/**
 * Interface for Common Success Modal.
 */
interface ISuccessModal {
  visible: boolean;
  footer?: any;
  onCancel: () => void;
  children: {};
  className?: string;
  centered?: boolean;
  width?: number | undefined;
  style?: {};
}

/**
 * This is used for SuccessModal
 *
 * @usage: <SuccessModal isModalVisible={visible} onCancel={function}/>
 */
const SuccessModal: FC<ISuccessModal> = (props: ISuccessModal) => {
  const { visible, footer, onCancel, children, className, centered = false, style, width= undefined } = props;

  return (
    <Modal
      visible={visible}
      footer={footer}
      maskClosable={true}
      onCancel={onCancel}
      className={`success-modal ${className}`}
      centered={centered}
      destroyOnClose={true}
      closable={false}
      width={width}
      style={style}
    >
      <div className="modal-content-wrapper">
        <Space direction={"vertical"} size={"middle"}>
          <CheckCircleFilled className="check-circle-icon" />
          {children}
        </Space>
      </div>
    </Modal>
  );
};

export default SuccessModal;
