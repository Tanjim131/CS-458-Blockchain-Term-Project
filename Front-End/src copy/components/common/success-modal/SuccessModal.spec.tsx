import React from 'react';
import { shallow } from "enzyme";
import SuccessModal from "./SuccessModal";
import { shallowToJson } from "enzyme-to-json";

describe('SuccessModal Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <SuccessModal
            visible = {true}
            footer = 'footer'
            onCancel = {mockFunc}
            children = "children"
            className = ""
            width = {undefined}
            style = {{}}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it("should check whether the text color is present", () => {
        wrapper.setProps({ centered : true, width: 120 });
        expect(wrapper).toBeTruthy()
    });

})