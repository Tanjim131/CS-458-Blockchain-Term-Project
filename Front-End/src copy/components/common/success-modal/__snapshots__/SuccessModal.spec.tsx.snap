// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal Component Check if component exists or not and capture snapshot 1`] = `
<Modal
  centered={false}
  className="success-modal "
  closable={false}
  confirmLoading={false}
  destroyOnClose={true}
  footer="footer"
  maskClosable={true}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[MockFunction]}
  style={Object {}}
  transitionName="zoom"
  visible={true}
  width={520}
>
  <div
    className="modal-content-wrapper"
  >
    <Space
      direction="vertical"
      size="middle"
    >
      <ForwardRef(CheckCircleFilled)
        className="check-circle-icon"
      />
      children
    </Space>
  </div>
</Modal>
`;
